### PR TITLE
fix(server): check if sidecarPath exists

### LINKED ISSUE #6169: Automatically clean up removed sidecar files
Immich supports adding and editing .XMP sidecar files in the library folder, and has a few jobs to discover and sync those. However deleting those files externally is currently not supported, as once a sidecar path has been added to the database, it is expected to stay on the filesystem as well. The [sidecar discovery job](https://github.com/immich-app/immich/blob/main/server/src/domain/metadata/metadata.service.ts#L283-L298) should be updated so that when a `sidecarPath` exists in the database but not on disk, the path in the database is nulled out.

### PR DESCRIPTION
**Pull request by JW-CH**
Fixes #6169

### CODE REVIEW

File: server/src/domain/metadata/metadata.service.ts [hunk 1]
```diff
@@ -431,7 +432,18 @@ export class MetadataService {
...
-    const sidecarTags = asset.sidecarPath ? await this.repository.readTags(asset.sidecarPath) : null;
+
+    let sidecarTags = null;
+    if (asset.sidecarPath) {
+      // check if sidecar-file exists
+      if (fs.existsSync(asset.sidecarPath)) {
+        sidecarTags = await this.repository.readTags(asset.sidecarPath);
+      } else {
+        this.logger.error(
+          `Sidecar-File '${asset.sidecarPath}' for asset '${asset.id}' does not exist on the filesystem`,
```
Comment (JW-CH) [inline server/src/domain/metadata/metadata.service.ts:443]: Is an error/warning enough or does the db-field need to be nulled when not found?

Comment (danieldietzler) [inline server/src/domain/metadata/metadata.service.ts:443]: Don't import `fs` in domain/ files.

**Review by danieldietzler — CHANGES_REQUESTED** [review]
~~What is the point of all this? `exiftool.read` should just throw an error (which we catch in the repository `readTags` method)~~
Has been discussed on Discord

**Review by jrasm91 — DISMISSED** [review]
It seems like this is something better suited for the sidecar sync/discover job. Sync can probably be updated to null out the database in the case the file no longer exists.

**Comment by JW-CH** [comment]
TODO: Add valuable tests 🤣

**Comment by danieldietzler** [comment]
Hey @JW-CH, just wanting to check in... Would you like us to add some tests or do you plan on adding them yourself?

**Review by danieldietzler — APPROVED** [review]

**Review by jrasm91 — APPROVED** [review]

**Pull request merged**